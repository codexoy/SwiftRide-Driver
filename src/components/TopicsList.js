import React, { Fragment } from "react"
import { StyleSheet, View, Text, FlatList } from "react-native"
import EmptyState from "./EmptyState"
import { strings } from "../i18n"

export default class TopicsList extends React.Component {
	constructor(props) {
		super(props)
	}
	renderTopic = ({ item, index }) => {
		return (
			<View style={styles.topic} key={`topic${item.id}`}>
				<Text style={styles.topicText}>{item.title}</Text>
			</View>
		)
	}

	_renderEmpty = () => (
		<EmptyState
			image="lessons"
			text={strings("empty_topics")}
			style={styles.empty}
		/>
	)

	render() {
		return (
			<FlatList
				data={this.props.topics}
				renderItem={this.renderTopic}
				keyExtractor={item => `topic${item.id}`}
				ListEmptyComponent={this._renderEmpty}
			/>
		)
	}
}

const styles = StyleSheet.create({
	topic: {
		borderBottomColor: "lightgray",
		paddingTop: 6,
		paddingBottom: 6,
		borderBottomWidth: 1,
		marginLeft: 12
	},
	topicText: {
		alignSelf: "flex-start"
	},
	empty: {
		alignSelf: "center"
	}
})
