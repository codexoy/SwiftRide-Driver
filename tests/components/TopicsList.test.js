import React from "react"

import renderer from "react-test-renderer"

import TopicsList from "../../src/components/TopicsList"

describe("TopicsList", () => {
	test("view renders correctly", () => {
		const tree = renderer
			.create(
				<TopicsList
					topics={[
						{ id: 1, title: "test1" },
						{ id: 2, title: "test2" }
					]}
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
