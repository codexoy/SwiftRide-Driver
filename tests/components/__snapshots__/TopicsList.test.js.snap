// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicsList view renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  data={
    Array [
      Object {
        "id": 1,
        "title": "test1",
      },
      Object {
        "id": 2,
        "title": "test2",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "borderBottomColor": "lightgray",
            "borderBottomWidth": 1,
            "marginLeft": 12,
            "paddingBottom": 6,
            "paddingTop": 6,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          test1
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "borderBottomColor": "lightgray",
            "borderBottomWidth": 1,
            "marginLeft": 12,
            "paddingBottom": 6,
            "paddingTop": 6,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          test2
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
